src/wp-includes/post.php

```diff
@@ -6480,6 +6480,25 @@ function wp_delete_attachment_files( $post_id, $meta, $backup_sizes, $file ) {
 					$deleted = false;
 				}
 			}
+			// Check for alternate size mime types in the sizeinfo['sources'] array to delete.
+			if ( isset( $sizeinfo['sources'] ) && is_array( $sizeinfo['sources'] ) && 1 < sizeof( $sizeinfo['sources'] ) ) {
+				$index = 0;
+				foreach ( $sizeinfo['sources'] as $mime => $properties ) {
+					// Skip the first mime type which was already deleted.
+					if ( 0 === $index++ ) {
+						continue;
+					}
```

Comment:
Better: Assign `$sizeinfo['sources']` to a separate variable and then use `array_shift` to remove the first item. This way you don't need the extra `$index` count and condition.

Also: Are we 100% sure the first MIME type is always the original one? I think what would be safer here is to unset whatever was the original MIME type used in the root of `sizes`.